src/Microsoft.Health.Fhir.Core/Data/R4/search-parameters.json

```diff
@@ -45840,7 +45840,7 @@
       },
       {
         "definition" : "http://hl7.org/fhir/SearchParameter/Observation-value-date",
-        "expression" : "value.as(DateTime) | value.as(Period)"
+        "expression" : "value.as(dateTime) | value.as(Period)"
```

Comment:
This sounds like a parser bug too. `If the identifier cannot be resolved to a valid type identifier, the evaluator will throw an error.` http://hl7.org/fhirpath/N1/#as-type-specifier